graylog2-server/src/main/java/org/graylog2/lookup/adapters/CSVFileDataAdapter.java

```diff
@@ -175,22 +189,43 @@ private Map<String, String> parseCSVFile() throws IOException {
                     if (keyColumn < 0 || valueColumn < 0) {
                         throw new IllegalStateException("Couldn't detect column number for key or value - check CSV file format");
                     }
+                    if (next.length == 1 && StringUtils.isEmpty(next[0])) {
+                        LOG.debug("Skipping empty line in CSV adapter file [{}/{}].", name, config.path());
+                        continue;
+                    }
+
+                    final String value;
+                    final String key;
+                    try {
+                        key = next[keyColumn];
+                        value = next[valueColumn];
+                    } catch (IndexOutOfBoundsException e) {
+                        final String error = f("The CSV file contains invalid lines. Please check the file and ensure " +
+                                "that both key and value columns are present in all lines.", name, config.path());
+                        throw new IllegalStateException(error, e);
+                    }
+
                     if (config.isCaseInsensitiveLookup()) {
-                        newLookupBuilder.put(next[keyColumn].toLowerCase(Locale.ENGLISH), next[valueColumn]);
+                        newLookupBuilder.put(key.toLowerCase(Locale.ENGLISH), value);
                     } else {
-                        newLookupBuilder.put(next[keyColumn], next[valueColumn]);
+                        newLookupBuilder.put(key, value);
                     }
                 }
             }
         } catch (Exception e) {
             LOG.error("Couldn't parse CSV file {} (settings separator=<{}> quotechar=<{}> key_column=<{}> value_column=<{}>)", config.path(),
                     config.separator(), config.quotechar(), config.keyColumn(), config.valueColumn(), e);
             setError(e);
+            throw new IllegalStateException(e);
```

Comment:
Parsing errors were being silently ignored in the previously, because `clearError` on line 147 was clearing it out. Throwing the exception here properly reports the unrecoverable error to the UI:

![image](https://user-images.githubusercontent.com/3423655/218857102-09c934ef-552c-405f-b44f-dfb88d88bc5c.png)